### DISCUSSION TITLE
Is there any method to calculate the ecef coord of the left-up corner of a specified cell id?

### DISCUSSION THREAD

**Discussion by xztao** [google/s2geometry#326] [General]

**Comment by smcallis** [comment]
We don't directly have a concept of ECEF in the sense of S2Points being points on an ellipsoid.  Depending on how you define left-up you can use S2Cell::GetVertex to get the corner vertices of an S2Cell though.  Note these are just the closest representable points in double precision, and might not actually be on the boundary of the cell if that matters to you.

**Reply by xztao** [reply]
Thanks for your quick reply! I believe GetVertex would be helpful for me!

**Comment by jmr** [comment]
What are you really trying to do?

https://en.m.wikipedia.org/wiki/XY_problem

**Reply by xztao** [reply]
Sorry for the confusion if I made any. 
I have to use a black-box sdk which receives float-type coords. If ECEF coords is sent, numerical error occurs due to the large values. To resolve the problem, a proper solution is to split the earth surface into small blocks as what S2geo does. Then the present of a position is something like (cell_id, coord_offset). What I need is reduced to calculate the anchor(say up-left corner) ecef coord according to cell_id.

**Reply by smcallis** [reply]
You could just create an S2CellId(point) which will create the nearest level 30 cell, then use the center of that as your reference.  L30 cells are spaced at ~10mm so if that resolution is sufficient for you then you can just store the S2CellId, no extra information needed.

**Reply by xztao** [reply]
Thanks for your advice! It should work in ~10mm error level. I'll make a try
